### DISCUSSION TITLE
Support for Telegraph Sounder

### DISCUSSION THREAD

**Discussion by moseskfrost** [Vail-CW/vail_repeater_depricated#61] [General]
This thread is for the purpose of Telegraph sounder implementation to home-brewed Vail adapters.

**Comment by nealey** [comment]
Moses was asking about hooking a telegraph sounder up to the vail adapter. The adapter already plays tones for received signals, so it would just be a matter of hooking into that code to send a signal out to your sounder relay.

For this, you are going to first need a Vail adapter. Right now, you can get either a Seeeduino XIAO, or an Adafruit QT Py. The Adafruit is a little easier to work with (it has a reset button instead of requiring you to connect two pads with a wire) and is slightly cheaper right now in the US, but either one is fine.

Your circuit is going to look a lot like this one from a [lastminuteengineers.com tutorial](https://lastminuteengineers.com/one-channel-relay-module-arduino-tutorial/):

![relay circuit](https://lastminuteengineers.b-cdn.net/wp-content/uploads/arduino/wiring-one-channel-relay-module-with-arduino.png)

---

For the first step, I think it would make sense to write a program that just opens and closes the relay every second. Here's something that should work:

```c++
#define RELAY_PIN 6

void setup() {
  pinMode(RELAY_PIN, OUPUT);
}

void loop() {
  digitalWrite(RELAY_PIN, HIGH);
  delay(500);
  digitalWrite(RELAY_PIN, LOW);
  delay(500);
}
```

You should be able to flash this to your microcontroller with the Arduino IDE, but if you're not up for that, let me know, and I will build a firmware for you that you can flash with your file manager just like [flashing the Vail adapter firmware](https://www.youtube.com/watch?v=IgOdkUe5SMY).

That's a lot of information, I know. Please feel encouraged to reach out in this thread for any clarifying questions you have along the way to building the auto-clicker circuit I just described! Your questions will help other people who want to do something similar!

**Reply by nealey** [reply]
Here's what you'll need for this:

* A Seeduino XIAO M0 or Adafruit QT Py M0 (others can work but these are the ones I support)
* A tiny breadboard
* Some hookup wire
* If you buy a microcontroller that doesn't have pre-soldered pins: a soldering iron and some solder

Steps:

1. If your microcontroller doesn't have pins soldered on yet, solder the pins onto the microcontroller. The [Adafruit Header Soldering Tutorial](https://learn.adafruit.com/how-to-solder-headers) will walk you through this.
2. Put the microcontroller into the breadboard. You'll want each row of pins on one side of the breadboard, like in [this image](https://cdn-learn.adafruit.com/assets/assets/000/097/844/original/circuitpython_QT_Py_Essentials_Analog_In_bb.jpg). Since each half of the breadboard is disconnected from the other half, this keeps you from connecting the left half to the right half.
3. Using hookup wire, connect your relay to power, ground, and pin 6. Hopefully the relay is labeled! Let me know if you need help with this.
4. Set up the Arduino IDE up for whatever microcontroller you're using, using the online guide for setting up the Arduino IDE with that microcontroller (do a web search).
5. Paste in that program and upload it to the microcontroller.
6. You should hear your relay click open and closed every 500 milliseconds!

Let me know how it goes!
